code/harness/DiskContents.h

```diff
@@ -22,33 +22,33 @@ namespace fs_testing {
 class fileAttributes {
 public:
   struct dirent* dir_attr;
```

Comment:
You still leak `dir_attr` since you malloc it when you call `set_dir_attr`, but never free it. I would actually make this a "smart pointer" i.e a `shared_ptr<dirent>` so that you don't have to worry about memory management for it. There are a few other places in the code that use either `shared_ptr` or `unique_ptr` to simplify memory management. The beautify of smart pointers is that they handle all the ref counting and will free their contents when they go out of scope